src/l3-resources/sagemaker-integration/index.ts

```diff
@@ -0,0 +1,397 @@
+      "PushToMDBLambdaFunction",
+      {
+        handler: aws_lambda.Handler.FROM_IMAGE,
+        runtime: aws_lambda.Runtime.FROM_IMAGE, // Replace with your preferred runtime
+        role: this.pushLambdaFunctionRole, // Use a role defined elsewhere
+        code: lambda.Code.fromEcrImage(pushRepo), // Replace with your ECR image URI
+        environment: {
+          mongo_endpoint: props.mongoDBEndpoint, // Replace with your MongoDB endpoint
+          dbname: props.databaseName, // Replace with your database name
+        },
+        architecture: lambda.Architecture.X86_64,
+        memorySize: 1024,
+        timeout: cdk.Duration.seconds(300),
+      }
+    );
+    const sagemakerEPArn = `arn:aws:sagemaker:${cdk.Stack.of(this).region}:${
+      cdk.Stack.of(this).account
+    }:endpoint/${this.sagemakerEndpoint.endpointName}`;
+    // Create a Role for Event Bridge to invoke Lambda to read MongoDB events
+    this.pullLambdaFunctionRole = new iam.Role(this, "PullLambdaFunctionRole", {
+      assumedBy: new iam.ServicePrincipal("lambda.amazonaws.com"),
+      roleName: "PullLambdaFunctionRole", // Set your desired role name
+      managedPolicies: [
+        iam.ManagedPolicy.fromAwsManagedPolicyName(
+          "service-role/AWSLambdaBasicExecutionRole"
+        ),
+      ],
+      inlinePolicies: {
+        SageMakerEndpointInvocationPolicy: new iam.PolicyDocument({
+          statements: [
+            new iam.PolicyStatement({
+              effect: iam.Effect.ALLOW,
+              actions: ["sagemaker:InvokeEndpoint"],
+              resources: [sagemakerEPArn], // Replace with your SageMaker endpoint ARN
+            }),
+            new iam.PolicyStatement({
+              effect: iam.Effect.ALLOW,
+              actions: ["events:PutEvents"],
+              resources: [this.eventbusForSagemakerResults.eventBusArn], // Replace with your event bus ARN
+            }),
+          ],
+        }),
+      },
+    });
+
+    const pullRepo = ecr.Repository.fromRepositoryName(
+      this,
+      "PullRepository",
+      props.pullLambdaImageRepo
+    );
+    // const pullRepo = new ecr.Repository(this, "PullRepository", {
+    //   repositoryName: props.pullLambdaImageURI,
+    // });
+    const endpointName = this.sagemakerEndpoint.endpointName ?? "";
+    // Create a Lambda function to pull results from SageMaker
+    this.lambdaFunctionToReadMDBEvents = new lambda.Function(
+      this,
+      "PullFromSageMakerLambdaFunction",
+      {
+        runtime: aws_lambda.Runtime.FROM_IMAGE,
+        code: lambda.Code.fromEcrImage(pullRepo),
+        handler: aws_lambda.Handler.FROM_IMAGE,
+        role: this.pullLambdaFunctionRole,
+        environment: {
+          model_endpoint: endpointName, // Use an existing resource or replace with your model endpoint
+          region_name: cdk.Aws.REGION,
+          eventbus_name: this.eventbusForSagemakerResults.eventBusName, // Use an existing resource or replace with your event bus
```

Comment:
I am not getting it the purpose of these comments. Should these be params then with a default value?

Reply:
agreed @marcosuma , since we are creating with in the CDK that comment is irrelevant, hence i removed.
Thanks